first_bioinf_tools.py

```diff
@@ -1,71 +1,86 @@
-    if function == "is_nucleic_acid":
-        for sequence in sequences[:-1]:
-            result = rna_dna.is_nucleic_acid(sequence)
-            output.append(result)
-
-    elif function == "transcribe":
-        for sequence in sequences[:-1]:
-            result = rna_dna.transcribe(sequence)
-            output.append(result)
-
-    elif function == "reverse":
-        for sequence in sequences[:-1]:
-            result = rna_dna.reverse(sequence)
-            output.append(result)
-
-    elif function == "complement":
-        for sequence in sequences[:-1]:
-            result = rna_dna.complement(sequence)
-            output.append(result)
-
-    elif function == "reverse_complement":
-        for sequence in sequences[:-1]:
-            result = rna_dna.reverse_complement(sequence)
-            output.append(result)
-
-    if len(output) == 1:  # Return output
+    for sequence in sequences:
+        num_of_seq += 1
+        if dict_functions[function](sequence) == None:
+            print(f"Error in {num_of_seq} sequence :{sequence}")
+        else:
+            output.append(dict_functions[function](sequence))
+    
+    # Return output
+    if len(output) == 1:
         return output[0]
     else:
         return output
 
-
-def filter_fastq(
-    seqs, gc_bounds=(0, 100), length_bounds=(0, 2**32), quality_threshold=0
-):
-
-    filtered_seqs = {}
-
-    if type(gc_bounds) == tuple:
-        gc_bounds_min = gc_bounds[0]
-        gc_bounds_max = gc_bounds[1]
-    else:
-        gc_bounds_min = 0
-        gc_bounds_max = gc_bounds
-
-    if type(length_bounds) == tuple:
-        length_bounds_min = length_bounds[0]
-        length_bounds_max = length_bounds[1]
-    else:
-        length_bounds_min = 0
-        length_bounds_max = length_bounds
-
-    filtered_seqs = fastq.gc_counting(seqs, gc_bounds_min, gc_bounds_max)
-
-    filtered_seqs = fastq.length_counting(
-        filtered_seqs, length_bounds_min, length_bounds_max
-    )
-
-    filtered_seqs = fastq.quality_counting(filtered_seqs, quality_threshold)
-
-    return filtered_seqs
+def filter_fastq(input_fastq, output_fastq, gc_bounds=(0, 100), length_bounds=(0, 2**32), quality_threshold=0):
+    
+    """
+    
+    The function allows you to filter reads by GC-content, length, and average quality.
+    
+    Arguments:    
+    input_fastq - The path to input FASTQ-file: str.
+    output_fastq - The path to output FASTQ-file: str.
+    gc_bounds - An interval of GC-content (in percent) for filtering (default is `(0, 100)`); if set as a single number, it defines the upper bound: int \ tuple.
+    length_bounds - An interval of length for filtering (default is `(0, 2**32)`); if set as a single number, it defines the upper bound: int \ tuple.
+    quality_threshold - A threshold of mean quality (phred33 scale) (default is `0`): int.
+    
+    """
+    
+    seq = {}
+    number_read = 0
+    end_file = False
+    
+    while not(end_file):
```

Comment:
```suggestion
    while not end_file:
```
Читается легче, смысл тот же.